### DISCUSSION TITLE
Ubisys S1 physical button sometimes not working

### DISCUSSION THREAD

**Discussion by Sailing74** [Koenkk/zigbee2mqtt#22983] [Devices]
I have quite a few S1 here. Like them, working very well with z2m. With two of the S1, however, I have a small problem with the physical button connected: Sometimes when I (or another family member) pushes the physical (momentary) button, the S1 doesn't turn on the power. In such case, pushing the button again does not help. BUT waiting appr. 5 secs and pushing the button then again does work.

I don't think that this is directly related to z2m, but maybe rather to the configuration or some other aspect of the S1...?

I have configured all S1 like this: 

'{"configure_device_setup": { "input_action_templates": { "type": "toggle" }}}'

Anything else I could try to fix configuration of the two S1 showing this behavior?

Thanks!

**Comment by sjorge** [comment]
How recently did you configure the two having problems?

There were some bugs that got recently fixed.

If it was recent it might be worth setting them to something else like dimmer_single and then back to toggle.

**Comment by Sailing74** [comment]
The initial configuration was with something in the range of 1.35 of z2m, so not too long ago. The well working S1s are much (well, months... but quite some months) older, so configured with an earlier version (1.31 or 1.32, if I had to guess). 

I gave it a try to reconfigure the S1, but was at max mildly successful. Alternative configurations I sent via mosquitto:

`'{"configure_device_setup": { "input_action_templates": { "type": "dimmer_single" }}}'`

and

`'{"configure_device_setup": { "input_action_templates": { "type": "on_off_switch" }}}'
`
All I'm getting is timeouts:

`[2024-06-10 17:19:01] error:    z2m: Publish 'set' 'configure_device_setup' to 'Lt_Bad_Spiegel' failed: 'Error: ZCL command 0x001fee0000006d24/232 manuSpecificUbisysDeviceSetup.writeStructured([{"attrId":1,"selector":{},"dataType":72,"elementData":{"elementType":"octetStr","elements":[[0,13,2,6,0,2]]}}], {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 23004 - 232 - 33 - 64512 - 16 after 10000ms)'`

BUT: Now the toggle switch (in the wall, the physical one) is not working anymore at all. Only switching via the z2m frontend works.

I'm getting the same timeouts when trying to configure the device again as a toggle.

Since the error is a timeout, you may ask about LQI: The frontend shows LQI as 80 currently. The map shows that the S1 has 10 connections in the network, ranging from 237/229 (to a Ubisys J1) to 1/129 (to the coordinator).

Any suggestion what I could try next, please?

**Reply by sjorge** [reply]
Just to double check, you are on 1.38.0 right?

You probably need to power-cycle the device, which can be a bit annoying when they are in the wall unfortunately.

**Reply by Sailing74** [reply]
I killed power through the breaker, let it sit for a minute, and turned power on again. No change, the button switch is still not working, and I'm still getting timeouts trying to configure the S1.

Should I try to factory reset it? (I'd try with the power-off-power-on sequence first, hoping that I don't have to get it out of the wall again...) And if so, I guess I should also delete it from z2m and then let it join again?

Or is there an easier way I could try first?

**Reply by sjorge** [reply]
I think if you enable joining and tell the device ti leave with z2m it should reset and rejoin. Not 100% sure on that though.

**Reply by Sailing74** [reply]
Tried that. The S1 left and rejoined, not change in behavior/usability - other than that z2m now shows the S1 as being "Offline" permanently.

Will try to reset it through some other mean and report back.

**Reply by sjorge** [reply]
@Sailing74 out of interest, are you also using ezsp based adaptor like @chris-1243  ?
The last fix was because with the zh refactor some Zcl.DataType's were not corrected update.
But that big rewrite also improved WriteStructure support in zh, I wonder if some stuff for that is missing or incorrect in ezsp vs zstack.

Before I was writing junk and the device was also ignoring it.

**Reply by sjorge** [reply]
OK so it got fixed here https://github.com/Koenkk/zigbee-herdsman-converters/pull/7588, which did not make it into 1.38.0 ...

Is either of you able to test the dev branch, that should include the writeStructure fix.

**Reply by Sailing74** [reply]
I have a CC2652P based Sonoff coordinator here.

As to testing: Is it as simple as stopping z2m, then
`npm install zigbee-herdsman und npm install zigbee-herdsman-converters `
and then starting z2m again?

**Reply by sjorge** [reply]
It's a bit more complicated than that: https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html#switch-to-the-dev-branch

**Reply by chris-1243** [reply]
I may try both adapter `zstack` and `ember` as my main network is on TI chip and I have a spare EZSP for testing...

Via Docker it is really easy to use the dev-branch. I'll do that at the end of my work day

@sjorge Any debug (herdsmann) log required if something goes wrong?

**Reply by Sailing74** [reply]
> It's a bit more complicated than that: https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html#switch-to-the-dev-branch

It's actually even more complicated, as this documentation does not lead to where it should be for Linux installs following the documentation: https://github.com/Koenkk/zigbee2mqtt/discussions/17938

But I think I got there and am (according to "about" in the frontend) now on:
```
1.38.0 commit: [f1847301](https://github.com/Koenkk/zigbee2mqtt/commit/f1847301)
[...]Frontend version
0.7.2
19.47.1
0.49.2
```
Is this good? (19.47.1 does not look quite like 19.48.0 as mentioned in https://github.com/Koenkk/zigbee-herdsman-converters/pull/7588 )?

Edit: Apparently not good enough. Still getting timeouts trying to configure the S1.

**Reply by sjorge** [reply]
I think that is the normal 1.38.0 release, IIRC it has `-dev` suffix on the z2m version.

**Reply by Sailing74** [reply]
Ok, redid everything, and appear to be on dev now. Good news:

`[2024-06-11 16:44:14] info: 	z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Lt_Bad_Spiegel', payload '{"configure_device_setup":{"input_configurations":[0]},"energy":0,"linkquality":65,"power":0,"state":"OFF","update":{"installed_version":37749810,"latest_version":37749810,"state":"idle"}}'`

Bad news: The (physical button) switch is still working intermittently, only. And if it doesn't work, pushing again doesn't help for about 5 secs. It appears the S1 hangs/waits for something/reboots... So maybe not related to the code changes...?

One thought I had was that if the physical button may sends a short signal or a few short signals, maybe this puts the S1 into a different state, waiting for something or rebooting. I did check the manual, but couldn't find something to this extent, though... Have you ever come across this?

Or any other idea, by chance?

**Reply by sjorge** [reply]
OK so with the `dev` branch you could write the config without a timeout, that's a good tart.

Mine respond pretty fast, I rebound mine to groups of lights not to the switch it self. The response is near instant.

**Reply by Sailing74** [reply]
Yes, the two older ones are great as well. Just the two newer ones, in 10% of the cases (and maybe when the button is not fully centrally hit at the initial attempt) take these breaks...
I can ask Ubisys support, but will need to supply them with the current configuration of the switch. If I set it to the action template "toggle", is this the same as applying the configuration shown in the technical reference in 7.7.5.2 as InputActions example for the S1-R(different device, but cannot find a momentary switch config for the S1 in the tecref) and momentary push buttons?

41 - element type: 0x41 (raw data)
02 00 - element count: 0x0002 (2 entries)
06 - element #1: six bytes
00 - InputAndOptions: 0x00
0D - Transition: released -> pressed
02 - Source: Endpoint #2 (hosts primary on/off client cluster on S1-R)
06 00 - Cluster ID: 0x0006 - on/off
02 - ZCL Command Template: Toggle

06 - element #2: six bytes
01 - InputAndOptions: 0x01
0D - Transition: released -> pressed
03 - Source: Endpoint #3 (hosts secondary on/off client cluster on S1-R)
06 00 - Cluster ID: 0x0006 - on/off
02 - ZCL Command Template: Toggle

Note: I don't think that this is correct, as the tecref states that for the S1 the InputAndOptions field may only be 0.

**Reply by sjorge** [reply]
I don't think the payload templating itself changed at all recently.

https://github.com/Koenkk/zigbee-herdsman-converters/blob/e204850a4ed41eb5eb85d65f5ae63b81ab61ed60/src/devices/ubisys.ts#L378

Is were the templates get translated. The ones I've used (toggle, dimmer_single, on_off_switch, and cover) all worked for me.

**Reply by chris-1243** [reply]
@sjorge 

> OK so with the dev branch you could write the config without a timeout, that's a good start.

I had a try on `ember` and `zstack` and there was no error at writing a configuration on my C4.

**Reply by sjorge** [reply]
Great, the config is also working as expected for you?

**Reply by chris-1243** [reply]
Yes. The configuration was taken by the device without error and while bound to a lamp, it was working as expected. Unfortunately, I still encounter the issue related in this topic #22766.

Regarding the issue on the C4, I suggest to continue in the topic mentioned above

**Reply by sjorge** [reply]
I'm not really sure how to fix that, if z2m doesn't get the message it doesn't known the command happened.

At least I'm glad the next release should have finally fixed the configuration.

**Reply by chris-1243** [reply]
I have a S2 configured the same (dimmer_single) and all commands are sent to the coordinator.. This is really strange

**Comment by chris-1243** [comment]
Sorry to bump on your thread, I just had a try on a test network with a C4 and I got exactly the same error message than @Sailing74. I used Z2M 1.38.0 and the C4 was paired only for the test.

[log.txt](https://github.com/user-attachments/files/15786026/log.txt)
